src/migrations/1679669193721-migrations.ts

```diff
@@ -0,0 +1,1612 @@
+                "deleted_at" TIMESTAMP,
+                CONSTRAINT "PK_ccc0ff5b7c575e40d57b48e77b8" PRIMARY KEY ("id"))
+                `);
+    await queryRunner.query(
+      'INSERT INTO "item" (id, name, type, description, path, creator_id, extra, settings, created_at, updated_at) SELECT id, name, type, description, path, creator, extra, settings, created_at, updated_at from item_old',
+    );
+    await queryRunner.query(`UPDATE "item" as a1 SET deleted_at = (
+                        SELECT created_at FROM recycled_item_old WHERE a1.path = recycled_item_old.item_path)`);
+
+    await queryRunner.query(`CREATE TABLE "item_membership" (
+            "id" uuid NOT NULL DEFAULT uuid_generate_v4(), 
+            "permission" character varying(100) NOT NULL,
+            "item_path" ltree NOT NULL REFERENCES item("path") ON DELETE CASCADE ON UPDATE CASCADE,
+            "creator_id" uuid REFERENCES member("id") ON DELETE SET NULL,
+            "member_id" uuid REFERENCES member("id") ON DELETE CASCADE,
+            "created_at" TIMESTAMP NOT NULL DEFAULT now(), 
+            "updated_at" TIMESTAMP NOT NULL DEFAULT now(),
+            CONSTRAINT "PK_ccc0ff5b7c575e40d57b48e77a8" PRIMARY KEY ("member_id", "item_path")
+            )`);
+    await queryRunner.query(
+      'ALTER TABLE "item_membership" ADD CONSTRAINT "item_membership-item-member" UNIQUE ("item_path", "member_id")',
+    );
+    await queryRunner.query(
+      `INSERT INTO "item_membership" (id,item_path, member_id, creator_id, permission, created_at, updated_at) 
+      SELECT id,item_path, member_id, creator, permission, created_at, updated_at from item_membership_old`,
+    );
+
+    // member password
+    await queryRunner.query(
+      'CREATE TABLE "member_password" ("id" uuid NOT NULL DEFAULT uuid_generate_v4(), "password" character varying(100) NOT NULL, "created_at" TIMESTAMP NOT NULL DEFAULT now(), "updated_at" TIMESTAMP NOT NULL DEFAULT now(), "member_id" uuid, CONSTRAINT "member-password" UNIQUE ("member_id"), CONSTRAINT "REL_81ea11a0e4f243edf76d53c284" UNIQUE ("member_id"), CONSTRAINT "PK_ff1a1183c81e78eaaa038f05a35" PRIMARY KEY ("id"))',
+    );
+
+    await queryRunner.query(
+      'ALTER TABLE "member_password" ADD CONSTRAINT "FK_81ea11a0e4f243edf76d53c2843" FOREIGN KEY ("member_id") REFERENCES "member"("id") ON DELETE CASCADE ON UPDATE NO ACTION',
+    );
+    await queryRunner.query(
+      'INSERT INTO "member_password" (password, member_id) SELECT password, id FROM member_old where password IS NOT NULL',
+    );
+
+    // recycled item
+    await queryRunner.query(
+      'CREATE TABLE "recycled_item_data" ("id" uuid NOT NULL DEFAULT uuid_generate_v4(), "created_at" TIMESTAMP NOT NULL DEFAULT now(), "creator_id" uuid NOT NULL, "item_path" ltree NOT NULL REFERENCES item("path") ON DELETE CASCADE ON UPDATE CASCADE, CONSTRAINT "recycled-item-data" UNIQUE ("item_path"), CONSTRAINT "PK_d6f781e5054e98174c35c87c225" PRIMARY KEY ("id"))',
```

Comment:
```suggestion
      'CREATE TABLE "recycled_item_data" ("id" uuid NOT NULL DEFAULT uuid_generate_v4(), "created_at" TIMESTAMP NOT NULL DEFAULT now(), "creator_id" uuid, "item_path" ltree NOT NULL REFERENCES item("path") ON DELETE CASCADE ON UPDATE CASCADE, CONSTRAINT "recycled-item-data" UNIQUE ("item_path"), CONSTRAINT "PK_d6f781e5054e98174c35c87c225" PRIMARY KEY ("id"))',
```

a lot of `creator` field in the original `recycled_item` table is NULL on production, the transfer of data between the two table fail because of this constraint